go/vt/topo/vschema.go

```diff
@@ -20,34 +20,58 @@ import (
 	"context"
 	"path"
 
-	"google.golang.org/protobuf/proto"
-
 	"vitess.io/vitess/go/vt/log"
 	"vitess.io/vitess/go/vt/vterrors"
 
 	vschemapb "vitess.io/vitess/go/vt/proto/vschema"
 )
 
-// SaveVSchema saves a Vschema. A valid Vschema should be passed in. It does not verify its correctness.
-// If the VSchema is empty, just remove it.
-func (ts *Server) SaveVSchema(ctx context.Context, keyspace string, vschema *vschemapb.Keyspace) error {
+// KeyspaceVSchemaInfo wraps a vschemapb.Keyspace and is a meta
+// struct that contains metadata to give the data more context
+// and convenience. This is the main way we interact with a
+// keyspace's vschema.
+type KeyspaceVSchemaInfo struct {
+	Name string
+	*vschemapb.Keyspace
+	version Version
+}
+
+func (k *KeyspaceVSchemaInfo) CloneVT() *KeyspaceVSchemaInfo {
```

Comment:
super nit: why did we name this a `CloneVT`?

Reply:
I wanted to override the embedded vschemapb.Keyspace CloneVT function. I'm fine renaming it to Clone e.g. though.